src/matchers/toHaveBeenCalledOnceWith.js

```diff
@@ -16,7 +16,9 @@ export function toHaveBeenCalledOnceWith(received, expected) {
   }
 
   const passOnce = received.mock.calls.length === 1;
-  const pass = passOnce && this.equals(expected, received.mock.calls[0][0]);
+  const mockCall = received.mock.calls[0];
+  const receivedArray = Array.isArray(mockCall) ? mockCall : [mockCall];
```

Comment:
I have checked `createToBeCalledWithMatcher` implementation in jest:
![image](https://user-images.githubusercontent.com/7756211/215685399-68e1ef1b-26cc-48a2-8e53-b633f6d8d53c.png)

And I've tried using `mockCall` directly without converting it to array and it seems to pass all your tests:

![image](https://user-images.githubusercontent.com/7756211/215685523-597d1a1b-8409-446a-b109-00645bdac336.png)